pkg/apis/pipeline/v1beta1/param_types.go

```diff
@@ -38,40 +38,64 @@ type ParamSpec struct {
 	// Name declares the name by which a parameter is referenced.
 	Name string `json:"name"`
 	// Type is the user-specified type of the parameter. The possible types
-	// are currently "string" and "array", and "string" is the default.
+	// are currently "string", "array" and "object", and "string" is the default.
 	// +optional
 	Type ParamType `json:"type,omitempty"`
 	// Description is a user-facing description of the parameter that may be
 	// used to populate a UI.
 	// +optional
 	Description string `json:"description,omitempty"`
+	// Properties is the JSON Schema properties to support key-value pairs parameter.
+	// +optional
+	Properties map[string]PropertySpec `json:"properties,omitempty"`
```

Comment:
OpenAPI defines properties as a recursive schema object - https://swagger.io/specification/#schema-object

Should we take a similar approach here with ParamSpec?

Reply:
IIUC, the tep proposes defaulting to string types for values. [section link](https://github.com/tektoncd/community/blob/main/teps/0075-object-param-and-result-types.md#defaulting-to-string-types-for-values)

> a. Only string types would initially be supported b. Even when other types are supported, the most common usage of objects will likely use string values

So we might not want to pursue the recursive schema object for now.

Reply:
Yeah I think this is also similar to the comment about whether we want to support nested objects for the initial stage.